src/content/tutorials/en/intro-to-p5-strands.mdx

```diff
@@ -0,0 +1,744 @@
+
+## What is a Shader?
+
+Shaders are GPU programs written in specialized languages like GLSL for WebGL. There are two main types:
+- **Vertex shaders:** Transform the vertices of shapes (defining where to draw something)
+- **Vertex shaders:** Determine the color of each pixel (defining how to draw something)
+
+<Callout title="Info">
+  A fragment shader is sometimes referred to as a pixel shader.
+</Callout>
+
+A vertex and a fragment shader are both required to render anything in WebGL mode. Fortunately, p5.js provides some default shaders that you can use, so you don't need to write your own shaders. Consider this sketch:
+
+<EditableSketch code={`
+let myShader;
+
+function lightingSetup() {
+  ambientLight(80, 80, 140);
+  const pointX = cos(millis() / 750) * 100;
+  const pointZ = sin(millis() / 750) * 100;
+  pointLight(80, 80, 255, pointX, -60, pointZ);
+}
+
+function setup() {
+  createCanvas(200, 200, WEBGL);
+}
+
+function draw() {
+  background(100);
+  lightingSetup();
+  // shader(baseMaterialShader());
+  sphere(80);
+}
+`} />
+
+Even without explicitly setting it, a shader calculates the lighting and shading of the fill, and another does the stroke. Uncommenting shader(baseMaterialShader()) produces the same result because p5.js uses this shader by default.
+
+<Callout title="Tip">
+Try changing the 2nd line in `draw` to use `baseColorShader()` instead. The blue tint of the light in `lightingSetup` is taken into account by the material shader.
+</ Callout>
+
+Alternatively, we can use the `baseColorShader` instead and the fill is now a solid white color, not affected by lighting. The filters available when you call `filter()` are also shaders which p5.js provides for you.
+
+Part of the fun of WebGL is that you can write your own shaders, and it unlocks a lot of possibilities for creating which would be difficult to with the p2D renderer. They can also run a lot faster. In the next section, we will start the main part of the tutorial and learn how to use the new p5.js shading language.
+
+## What is p5.strands?
+
+p5.strands takes its name from the way shaders process information in parallel, rather than sequentially.
+
+In the rest of p5.js, we're accustomed to writing instructions that run sequentially, one after another. We might call `circle(10, 10)` and tell the canvas, "Draw a white circle at point `(10, 10)`." We then continue giving instructions, layer by layer, drawing each shape on top of the previous ones.
+
+Instead of working sequentially, shaders apply a set of instructions across all pixels simultaneously. Each pixel or vertex independently follows the same rules, but produces a unique result based on its position. Rather than explicitly drawing the circle, we instead ask each pixel individually, "Are you within a circle centered at `(10, 10)`? If so, change your color."

```

Comment:
Minor: "across all pixels or vertices"

or (minor, style) maybe always keep the same order throughout the tutorial? First vertex then pixel/fragment. So the repetition helps to orient.

Reply:
+1 on "Rather than explicitly drawing the circle, we instead ask each pixel individually, "Are you within a circle centered at `(10, 10)`? If so, change your color."" That's wonderful, could even be highlighted somewhere else in the tutorial. If a reader gets 1 thing out of this, I think that would be a good candidate for that 1 takeaway.